### PR TITLE
Implement eth_submitHashrate and eth_submitWork

### PR DESCRIPTION
**Pull request by njgheorghita**
### What was wrong?
`eth_submitHashrate` needed implementation according to [ERC 1474](https://github.com/bitpshr/EIPs/blob/147609158191a701cf02c8556d5e529ffcb31cd2/EIPS/eip-1474.md#data)

Related to Issue #1273 

#### Cute Animal Picture
image

### CODE REVIEW

File: web3/_utils/rpc_abi.py [hunk 1]
```diff
@@ -51,6 +51,7 @@
     'eth_sendRawTransaction': ['bytes'],
     'eth_sendTransaction': TRANSACTION_PARAMS_ABIS,
     'eth_sign': ['address', 'bytes'],
+    'eth_submitHashrate': ['uint', None],
```
Comment (carver) [inline web3/_utils/rpc_abi.py]: I think this is right (not sure if it's always `bytes32`)
```suggestion
    'eth_submitHashrate': ['uint', 'bytes'],
```
Also, maybe `['bytes8', 'bytes32', 'bytes32']` for `eth_submitWork`?

Adding this extra fields should automatically handle arguments like submitting hex strings, hex literals, or python `bytes`, as well as length checks.

File: web3/middleware/pythonic.py [hunk 2]
```diff
@@ -287,6 +296,12 @@ def to_hexbytes(num_bytes, val, variable_length=False):
             (estimate_gas_with_block_id, is_length(2)),
         )),
         'eth_sendTransaction': apply_formatter_at_index(transaction_param_formatter, 0),
+        'eth_submitHashrate': apply_formatter_at_index(hexstr_if_str(to_hex), 1),
+        'eth_submitWork': compose(
+            apply_formatter_at_index(nonce_formatter, 0),
+            apply_formatter_at_index(hexstr_if_str(to_hex), 1),
+            apply_formatter_at_index(hexstr_if_str(to_hex), 2),
+        ),
```
Comment (carver) [inline web3/middleware/pythonic.py]: All these changes to pythonic can go away with the `rpc_abi` change, I think.

File: web3/_utils/rpc_abi.py (hunk 1 above)
Comment (kclowes) [reply web3/_utils/rpc_abi.py]: [geth](https://github.com/ethereum/wiki/wiki/JSON-RPC#eth_submithashrate) and [parity](https://wiki.parity.io/JSONRPC-eth-module#eth_submithashrate) both specify 32 bytes, but I don't think the EIP does  ¯\_(ツ)_/¯

Comment (carver) [reply web3/_utils/rpc_abi.py]: Eh, the EIP is still a draft. Since geth & parity implement the same way, that wins over the doc. :)

**Comment by njgheorghita** [comment]
Ready for final review

File: web3/_utils/rpc_abi.py [hunk 3]
```diff
@@ -51,6 +51,8 @@
     'eth_sendRawTransaction': ['bytes'],
     'eth_sendTransaction': TRANSACTION_PARAMS_ABIS,
     'eth_sign': ['address', 'bytes'],
+    'eth_submitHashrate': ['uint', 'bytes'],
```
Comment (kclowes) [inline web3/_utils/rpc_abi.py]: I think this line should be [bytes32, bytes32]?

Comment (njgheorghita) [reply web3/_utils/rpc_abi.py]: While parity's fine with that, it seems like geth doesn't like that, going with `['uint', 'bytes32']`

Comment (kclowes) [reply web3/_utils/rpc_abi.py]: Iiiinteresting. [geth's docs](https://github.com/ethereum/wiki/wiki/JSON-RPC#eth_submithashrate) are wrong then. Sounds good!

**Comment by kclowes** [comment]
I just merged an `Unreleased` section to `docs/releases.rst`. Before you merge, will you add a blurb there please? Thanks!

**Pull request merged**